ledger/src/shred.rs

```diff
@@ -436,6 +436,40 @@ impl Shred {
         self.common_header.version
     }
 
+    // Returns the block index within the erasure coding set.
+    fn erasure_block_index(&self) -> Option<usize> {
+        let fec_set_index = self.common_header.fec_set_index;
+        let index = self.index().checked_sub(fec_set_index)? as usize;
+        if self.is_data() {
+            Some(index)
```

Comment:
Should we also uphold this existing check:

```
if index >= num_data_shreds {
     return Err(InvalidIndex);
}
```

Reply:
We don't know number of data shreds here, and I don't want to pass it in as a function argument (or check it inside Shredder) because then that would leak out implementation details.
The goal of this commit was to encapsulate and hide the implementation details of erasure coding indices.

I think it is fine not to check for it because data reconstruct from codes will error out anyways.

The check was only added recently in https://github.com/solana-labs/solana/pull/21082
and is not released yet.